client/client_test.go

```diff
@@ -5728,6 +5729,115 @@ func testBuildInfoNoExport(t *testing.T, sb integration.Sandbox) {
 	require.Equal(t, exbi.Sources[0].Ref, "docker.io/library/busybox:latest")
 }
 
+func testPullWithLayerLimit(t *testing.T, sb integration.Sandbox) {
```

Comment:
Worth adding a quick test in here that setting the limit to 0 doesn't break anything, also some of the other corner cases like that an error is returned when limit is greater than layers in the image.

Reply:
Added testcase